src/execution_plan/ops/op_semi_apply.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2018-2019 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Redis Labs Source Available License Agreement
+ */
+
+#include "op_semi_apply.h"
+#include "../execution_plan.h"
+
+// Forward declarations.
+void SemiApplyFree(OpBase *opBase);
+OpResult SemiApplyInit(OpBase *opBase);
+Record SemiApplyConsume(OpBase *opBase);
+OpResult SemiApplyReset(OpBase *opBase);
+
+static Record _pullFromBranchStream(OpSemiApply *op) {
+	// Propegate record to the top of the match stream.
+	if(op->op_arg) Argument_AddRecord(op->op_arg, Record_Clone(op->r));
+	return OpBase_Consume(op->match_branch);
+}
+
+static Record OpSemiApply_SemiApplyConsume(OpBase *opBase) {
```

Comment:
I think removing the `OpSemiApply_` prefix would improve this function name.

Reply:
done